File: app/src/main/java/com/task/paginationlist/presentation/piclist/views/fragments/PictureFragment.java
package com.task.paginationlist.presentation.piclist.views.fragments;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.arellomobile.mvp.MvpAppCompatFragment;
import com.arellomobile.mvp.presenter.InjectPresenter;
import com.arellomobile.mvp.presenter.ProvidePresenter;
import com.task.paginationlist.PaginationListApplication;
import com.task.paginationlist.R;
import com.task.paginationlist.data.utils.ILoader;
import com.task.paginationlist.domain.interactors.WallpaperListInteractor;
import com.task.paginationlist.presentation.piclist.presenter.PicturePresenter;
import com.task.paginationlist.presentation.piclist.views.fragments.intefaces.IPictureView;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;


public class PictureFragment extends MvpAppCompatFragment implements IPictureView {

    public static final String PAGE_NUM = "page_num";
    public static final String PAGE_POSITION = "page_position";


    public static PictureFragment createPictureFragment(Pair<Integer, Integer> pair) {
        PictureFragment fragment = new PictureFragment();
        Bundle bundle = new Bundle();
        bundle.putInt(PAGE_NUM, pair.first);
        bundle.putInt(PAGE_POSITION, pair.second);
        fragment.setArguments(bundle);
        return fragment;
    }

    @BindView(R.id.full_picture)
    ImageView imageView;

    @Inject
    ILoader loader;

    @Inject
    WallpaperListInteractor interactor;

    @InjectPresenter
    public PicturePresenter presenter;

    @ProvidePresenter
    public PicturePresenter providePresenter() {
        return new PicturePresenter(interactor, getArguments().getInt(PAGE_NUM), getArguments().getInt(PAGE_POSITION));
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        PaginationListApplication.getComponent().inject(this);
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_picture, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void setImageUrl(String url) {
        loader.loadImage(url, imageView);
    }
}

File: app/src/main/java/com/task/paginationlist/presentation/piclist/views/fragments/PictureFragment.kt
package com.task.paginationlist.presentation.piclist.views.fragments

import android.os.Bundle
import android.support.v4.util.Pair
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView

import com.arellomobile.mvp.MvpAppCompatFragment
import com.arellomobile.mvp.presenter.InjectPresenter
import com.arellomobile.mvp.presenter.ProvidePresenter
import com.task.paginationlist.PaginationListApplication
import com.task.paginationlist.R
import com.task.paginationlist.data.utils.ILoader
import com.task.paginationlist.domain.interactors.WallpaperListInteractor
import com.task.paginationlist.presentation.piclist.presenter.PicturePresenter
import com.task.paginationlist.presentation.piclist.views.fragments.intefaces.IPictureView

import javax.inject.Inject

import butterknife.BindView
import butterknife.ButterKnife


class PictureFragment : MvpAppCompatFragment(), IPictureView {

    @BindView(R.id.full_picture)
    internal var imageView: ImageView? = null

    @Inject
    internal var loader: ILoader? = null

    @Inject
    internal var interactor: WallpaperListInteractor? = null

    @InjectPresenter
    var presenter: PicturePresenter? = null

    @ProvidePresenter
    fun providePresenter(): PicturePresenter {
        return PicturePresenter(interactor!!, arguments.getInt(PAGE_NUM), arguments.getInt(PAGE_POSITION))
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        PaginationListApplication.getComponent().inject(this)
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater!!.inflate(R.layout.fragment_picture, container, false)
        ButterKnife.bind(this, view)
        return view
    }

    override fun setImageUrl(url: String) {
        loader!!.loadImage(url, imageView!!)
    }

    companion object {

        val PAGE_NUM = "page_num"
        val PAGE_POSITION = "page_position"


        fun createPictureFragment(pair: Pair<Int, Int>): PictureFragment {
            val fragment = PictureFragment()
            val bundle = Bundle()
            bundle.putInt(PAGE_NUM, pair.first)
            bundle.putInt(PAGE_POSITION, pair.second)
            fragment.arguments = bundle
            return fragment
        }
    }
}
